modules/reindex/src/main/java/org/elasticsearch/reindex/ReindexValidator.java

```diff
@@ -138,12 +139,27 @@ static void validateAgainstAliases(
              */
             target = indexNameExpressionResolver.concreteWriteIndex(clusterState, destination).getName();
         }
-        for (String sourceIndex : indexNameExpressionResolver.concreteIndexNames(clusterState, source)) {
+        SearchRequest filteredSource = skipRemoteIndexNames(source);
+        if (filteredSource.indices().length == 0) {
+            return;
+        }
+        String[] sourceIndexNames = indexNameExpressionResolver.concreteIndexNames(clusterState, filteredSource);
+        for (String sourceIndex : sourceIndexNames) {
             if (sourceIndex.equals(target)) {
                 ActionRequestValidationException e = new ActionRequestValidationException();
                 e.addValidationError("reindex cannot write into an index its reading from [" + target + ']');
                 throw e;
             }
         }
     }
+
+    private static SearchRequest skipRemoteIndexNames(SearchRequest source) {
+        return new SearchRequest(source).indices(
+            Arrays.stream(source.indices()).filter(name -> isRemoteExpression(name) == false).toArray(String[]::new)
+        );
+    }
+
+    private static boolean isRemoteExpression(String expression) {
+        return expression.contains(":");
```

Comment:
small nit: you could have used the existing RemoteClusterAware.REMOTE_CLUSTER_INDEX_SEPARATOR constant here.

Reply:
TIL; @volodk85 would you make that change?

Reply:
If you don't mind I'll address this in follow up [PR](https://github.com/elastic/elasticsearch/pull/100646) (complex datemath things) which I opened yesterday for Search team.

Reply:
Sounds good :+1: